airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -121,6 +120,23 @@ def _get_stream_kwargs(self, config: ConnectorConfig) -> Mapping[str, Any]:
         }
         return stream_kwargs
 
+    def get_role(self, config: ConnectorConfig) -> dict:
+        boto3_client = boto3.client("sts", aws_access_key_id=config.aws_access_key, aws_secret_access_key=config.aws_secret_key)
+        *_, arn_resource = config.role_arn.split(":")
+        if arn_resource.startswith("user"):
+            role = self._get_session_token(boto3_client)
+        elif arn_resource.startswith("role"):
+            role = self._assume_role(boto3_client, config.role_arn)
+        else:
+            raise ValueError("Invalid User/Role Arn")
+        return role
+
+    def _get_session_token(self, boto3_client):
+        return boto3_client.get_session_token()
+
+    def _assume_role(self, boto3_client, role_arn: str):
+        return boto3_client.assume_role(RoleArn=role_arn, RoleSessionName="guid")
```

Comment:
I would suggest to not creating dedicated one-line functions and calling boto client's methods directly from `get_role`

Reply:
agree